### PR TITLE
Merge asset-swapper with erc20-bridge-sampler

### PR DESCRIPTION
**Pull request by dorothy-zbornak**
## Description

*Most of the file changes are just renames and deletions.*

This moves the `erc20-bridge-sampler` contracts into `packages/asset-swapper`. The `contracts/erc20-bridge-sampler` package is no more. This should mean 0x-API no longer has to gitpkg `contract-artifacts` and `contract-wrappers`.

Also removed the sampler contract's dependency on `DevUtils` by inlining the fillable taker amount check on native orders.

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
         "typeRoots": ["node_modules/@0x/typescript-typings/types", "node_modules/@types"],
         "strict": true,
         // These settings are required for TypeScript project references
-        "composite": true,
+        // "composite": true,
```
Comment (dorothy-zbornak) [inline tsconfig.json:14]: Get bizarre tsc compilation errors with this enabled.

File: tsconfig.json [hunk 2]
```diff
@@ -11,7 +11,6 @@
         "typeRoots": ["node_modules/@0x/typescript-typings/types", "node_modules/@types"],
         "strict": true,
         // These settings are required for TypeScript project references
-        "composite": true,
```
Comment (dorothy-zbornak) [inline tsconfig.json:14]: Get bizarre tsc build errors with this enabled. Doesn't seem to affect anything to leave it out?

File: packages/asset-swapper/contracts/src/NativeOrderSampler.sol [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+                orderFillableMakerAssetAmounts[i] = LibMath.getPartialAmountCeil(
+                    orderFillableMakerAssetAmounts[i],
+                    orders[i].takerAssetAmount,
+                    orders[i].makerAssetAmount
+                );
+            }
+        }
+    }
+
+    function getOrderFillableAmount(
```
Comment (dorothy-zbornak) [inline packages/asset-swapper/contracts/src/NativeOrderSampler.sol]: Adapted from `DevUtils`. Does not use `compatBalanceOf()`, etc, for compatibility with tokens that return extra data.

File: .prettierignore [hunk 4]
```diff
@@ -40,10 +40,10 @@ lib
...
 /contracts/erc20/generated-artifacts
 /contracts/erc20/test/generated-artifacts
-/contracts/erc20-bridge-sampler/generated-wrappers
-/contracts/erc20-bridge-sampler/test/generated-wrappers
-/contracts/erc20-bridge-sampler/generated-artifacts
-/contracts/erc20-bridge-sampler/test/generated-artifacts
+/packages/asset-swapper/generated-wrappers
+/packages/asset-swapper/test/generated-wrappers
+/packages/asset-swapper/generated-artifacts
+/packages/asset-swapper/test/generated-artifacts
```
Comment (moodlezoup) [inline .prettierignore]: duplicated below

File: packages/asset-swapper/contracts/src/NativeOrderSampler.sol [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-erc20/contracts/src/interfaces/IERC20Token.sol";
+import "@0x/contracts-exchange/contracts/src/interfaces/IExchange.sol";
+import "@0x/contracts-exchange-libs/contracts/src/LibOrder.sol";
+import "@0x/contracts-exchange-libs/contracts/src/LibMath.sol";
+import "@0x/contracts-utils/contracts/src/LibBytes.sol";
+import "@0x/contracts-utils/contracts/src/LibSafeMath.sol";
+import "@0x/contracts-utils/contracts/src/DeploymentConstants.sol";
+import "./IDevUtils.sol";
```
Comment (moodlezoup) [inline packages/asset-swapper/contracts/src/NativeOrderSampler.sol]:
```suggestion
```
We can also delete `IDevUtils` itself

File: packages/asset-swapper/contracts/src/NativeOrderSampler.sol [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+
+contract NativeOrderSampler is
+    DeploymentConstants
+{
+    using LibSafeMath for uint256;
+    using LibBytes for bytes;
+
+    /// @dev The Exchange ERC20Proxy ID.
+    bytes4 private constant ERC20_ASSET_PROXY_ID = 0xf47261b0;
+    /// @dev Gas limit for DevUtils calls.
```
Comment (moodlezoup) [inline packages/asset-swapper/contracts/src/NativeOrderSampler.sol]:
```suggestion
    /// @dev Gas limit for staticcalls.
```

File: packages/asset-swapper/contracts/src/NativeOrderSampler.sol [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+import "@0x/contracts-exchange-libs/contracts/src/LibOrder.sol";
+import "@0x/contracts-exchange-libs/contracts/src/LibMath.sol";
+import "@0x/contracts-utils/contracts/src/LibBytes.sol";
+import "@0x/contracts-utils/contracts/src/LibSafeMath.sol";
+import "@0x/contracts-utils/contracts/src/DeploymentConstants.sol";
+import "./IDevUtils.sol";
+
+
+contract NativeOrderSampler is
+    DeploymentConstants
```
Comment (moodlezoup) [inline packages/asset-swapper/contracts/src/NativeOrderSampler.sol]: we also no longer need `DeploymentConstants`

**Review by moodlezoup — APPROVED** [review]
one step closer to deleting DevUtils ✊

File: packages/asset-swapper/contracts/src/NativeOrderSampler.sol [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+        pure
+        returns (IERC20Token token)
+    {
+        if (assetData.length == 0) {
+            return IERC20Token(address(0));
+        }
+        if (assetData.length != 36 ||
+            assetData.readBytes4(0) != ERC20_ASSET_PROXY_ID)
+        {
+            revert('NativeOrderSampler/UNSUPPORTED_ASSET_DATA');
```
Comment (dekz) [inline packages/asset-swapper/contracts/src/NativeOrderSampler.sol]: can we do `address(0)` and return `0` if it's unsupported? Always concerned about a bad order getting in and messing the whole sampler result.

**Review by dekz — APPROVED** [review]

**Pull request merged**